doc/source/extended_documentation/calibration/rainforests_calibration/rainforests_calibration.rst

```diff
@@ -0,0 +1,173 @@
+dataframe.
+
+Using this dataframe, a series of lightGBM binary-classification models are trained against
+the truth values for exceedances of the forecast error relative to a series of error-threshold
+values, resulting in one model per error threshold.
+
+As the calibration method works on a per realization basis, the errors used in training
+must be representative of the underlying systematic model biases for a single ensemble
+member rather than those of the ensemble as a whole. This requires the use of a single
+representative ensemble forecast member in training. It also requires the underlying
+weather type associated with the forecast and observation in the forecast-observation
+pair be as consistent as possible.
+
+For the forecast values used in training, we use the control at the shortest available
+lead-times that distinctly represent of the underlying weather types. Specifically, we
+take the earliest 24-hour period for daily accumulations, and all lead-times over the
+earliest 24-hour period for sub-daily accumulations to capture the full cycle of diurnal
+variations.
+
+Currently model training is done offline, using a minimum 12-month period to capture the
+full seasonal cycle.
+
+===========================
+Forecast calibration
+===========================
+
+Forecast calibration uses the trained GBDT models, along with the forecast cube and associated
+feature cubes. The tree-models are passed in via a model-config json which identifies
+the appropriate tree-model file for each error-threshold.
+
+Forecast calibration proceeds via a 2-step process:
+
+1. Evaluate the error CDF defined over the series of error-thresholds used in model training.
+  Each exceedance probability is evaluated using the corresponding tree-model, and the feature
+  variables as inputs.
+
+3. Interpolate the CDF to extract a series of percentiles over for the error distributions.
+  The error percentiles are then added to each associated ensemble realization from the
+  forecast variable to produce a series of realisable forecast values.
+
+Collectively these series form the calibrated super-ensemble which is obtained by collapsing
+the two realization dimensions into one. This is then sampled to provide the calibrated
+ensemble forecast.
+
+Deterministic forecasts can also be calibrated using the same approach to produce a calibrated
+pseudo-ensemble; in this case inputs are treated as an ensemble of size 1.
+
+** A typical usage example: ** we typically use around 25 error threshold values to construct
+the CDF for the distribution of forecast errors. For each error threshold we have an associated
+GBDT model which is used to evaluate the exceedance probabilities that describe the CDF.
+So starting with an input ensemble forecast consisting of 50 realizations, we evaluate 25
+threshold probability values for each realization to construct a forecast error CDF for each
+realization (50 in total).
```

Comment:
The intention was to spell out that there are 50 error distributions, but as it stands it is not entirely clear. Maybe this could become
```suggestion
realization (50 distributions in total, each containing 25 threshold values).
```